### DISCUSSION TITLE
MANUAL updating of deprecated, missing, incomplete or non-standard tags

### DISCUSSION THREAD

**Discussion by chrisdebian** [Helium314/SCEE#451] [Ideas]
Hi, everybody.

I'd like to start a discussion about the MANUAL updating of deprecated, missing, incomplete or non-standard tags. Streetcomplete and its contributors are very good at adding missing information, but sometimes information can't be added, because a tag may be deprecated or incomplete. In this instance, the replacement tag is likely to have more tagging options, which would mean that if the tag was updated from the current (deprecated) tag to the new tag, then we could potentially add more data.

Most readers will be aware that automated mass-edits are strongly discouraged, and I understand this; but, I still feel that deprecated/ incomplete tags need to be considered.

A list of deprecated tags can be found [here](https://wiki.openstreetmap.org/wiki/Deprecated_features#Deprecated_features), and this includes Overpass queries.

![image](https://user-images.githubusercontent.com/17935223/227197935-98b3c3d1-2fb3-4908-899a-634b551dfe55.png)

I'm not convinced that this is the definitive list of deprecated tags, as looking at a random area in the UK, I can [see](https://www.openstreetmap.org/edit?relation=146656#map=19/53.47263/-2.18065) (using the ID editor/ click on 'Issues (warning triangle)' on the right hand edge) that other issues exist, such as 'xx has no descriptive tags', 'Marked Crossing has incomplete tags' and 'Petrol Station has non-standard tags'.

![image](https://user-images.githubusercontent.com/17935223/227197223-675fb7b3-77e9-4a3a-ba37-c4342341fa1a.png)

[This page](https://wiki.openstreetmap.org/wiki/Category:Tag_descriptions_with_status_%22deprecated%22) may be helpful.

![image](https://user-images.githubusercontent.com/17935223/227199467-371fe405-c861-4f17-9ebf-0466dca9f9f3.png)

Another resource is KeepRight, which can identify missing tags, misspelled tags and deprecated tags. [Here](https://keepright.at/report_map.php?zoom=14&lat=48.20808&lon=16.37221) is an example; you can select the data you want to see, on the left side of the screen.

![image](https://user-images.githubusercontent.com/17935223/227197531-ccc94a75-ab39-451d-9bf0-4873f1fa760b.png)

I would welcome your thoughts on whether submitting a formal request for a Quest would be sensible. This would probably not be a good fit for Streetcomplete, but may be a possibility in SCEE.

Many thanks,

Chris
chris_debian
UK

**Comment by Helium314** [comment]
> A list of deprecated tags can be found [here](https://wiki.openstreetmap.org/wiki/Deprecated_features#Deprecated_features), and this includes Overpass queries.

Some of these deprecated features (e.g. `emergencycy=dry_riser_inlet`) already trigger Osmose issues, which then would need to be solved by manually editing tags.
Manual solving is always necessary in current state of SCEE, because Osmose fixables can't be mass-downloaded, see https://github.com/osm-fr/osmose-frontend/issues/404. But in the example of `emergencycy=dry_riser_inlet`, there is no fixable anyway (though a solution is in the description).

A quest would be more effort, and risk that users just answer with the suggested solution. Though this could be avoided, e.g. by using a safe subset of deprecated tags and replacements by default, and allow users to select more deprecated tags in quest settings.

> I'm not convinced that this is the definitive list of deprecated tags, as looking at a random area in the UK, I can [see](https://www.openstreetmap.org/edit?relation=146656#map=19/53.47263/-2.18065) (using the ID editor/ click on 'Issues (warning triangle)' on the right hand edge) that other issues exist, such as 'xx has no descriptive tags', 'Marked Crossing has incomplete tags' and 'Petrol Station has non-standard tags'.

iD sometimes shows warnings where others see everything as fine, e.g. `crossing=unmarked` triggers the "incomplete tags" warning, but the necessity of adding `crossing:markings=no` is not seen by everyone.
Other of those warnings like "no descriptive tags" are also found via Osmose issues.

> Another resource is KeepRight, which can identify missing tags, misspelled tags and deprecated tags. [Here](https://keepright.at/report_map.php?zoom=14&lat=48.20808&lon=16.37221) is an example; you can select the data you want to see, on the left side of the screen.

I considered adding a quest for KeepRight issues, but didn't find a way to report errors as corrected or false positive via API. Probably there is a way as it can be done on the website, but I didn't want to spend much time with that.

> I would welcome your thoughts on whether submitting a formal request for a Quest would be sensible. This would probably not be a good fit for Streetcomplete, but may be a possibility in SCEE.

Adding such a quest seems resonable, but I guess some cases will need better resolution than semi-automated tag updates.

**Comment by mnalis** [comment]
Adding keepright in addition to osmose checks seems like a reasonable option to me. But that requires expert mode for manual editing of tags, of course. Advantage is that user if in full control, and should check all tags for each element and make separate decision in each case, and that process works for heuristically detected "misspellings" as well as unknown and/or deprecated tags. 

(In the meantime, if you're interested in QA on mobile, [OsmBugs](https://f-droid.org/packages/org.gittner.osmbugs/) app supports  OpenStreetMap Notes + Osmose +  Mapdust + Keepright).

---

As for the specific issue of deprecated tags, and what to do about them less manually, one would have to be careful:

Are those _**really**_ deprecated tags, or someone has just edited them on the wiki and said so (or added check in Osmose, which is not without fault either), which happens annoyingly often? IOW, would **anybody** ever complain if those tags were replaced? If everybody is agreement that that old tag should be replaced with something else, then :

- if the tag is **deprecated because it was ambiguous**, then a specific quest can be created (e.g. was that `amenity=dog_waste_bin`  intended to be `amenity=waste_basket + waste=dog_excrement` _**or**_ `amenity=vending_machine + vending=excrement_bags`  _**or**_ both _**or**_ neither) ? Of course, it would probably only make sense to invest time for quests that have reasonable number of occurrences in the wild (e.g. that specific `amenity=dog_waste_bin` tag has 1 use worldwide, so is probably not an ideal candidate for writing such new quest :rofl: )

- if the tag was deprecated because there is existing tag with **exactly the same** unambiguous meaning, and no on-the-ground verification is needed to confirm that (like e.g. converting `emergency=aed` to `emergency=defibrillator`), then I think it **should be done via automated edit** - yes, it needs some effort to write the automated edit procedure, but at least it does not pointlessly waste valuable mapper time just to _"rubberstamp"_ the change. Automated edit is much preferred not only because it saves on-the-ground-mapper time, but also people might complain (which might mean it should've not been deprecated in the first place!), and even if the problem with _"fixing deprecated tag"_ is detected too late, it is much easier to stop the bot and revert few automated edits then to track thousands of users and convince them to upgrade first so cleanup can proceed).
